Sources/App/Controllers/FormatAPIController.swift

```diff
@@ -0,0 +1,222 @@
+                throw Abort(.badRequest, reason: "Unknown locale: '\(localeID)'")
+            }
+            locale = Locale(identifier: actualLocaleID)
+        } else {
+            locale = Locale(identifier: "en_US_POSIX")
+        }
+        
+        if let tzID = request.timeZone {
+            let loweredTZID = tzID.lowercased()
+            guard let actualTZID = TimeZone.knownTimeZoneIdentifiers.first(where: { $0.lowercased() == loweredTZID }) else {
+                throw Abort(.badRequest, reason: "Unknown time zone: '\(tzID)'")
+            }
+            timeZone = TimeZone(identifier: actualTZID)!
+        } else {
+            if #available(macOS 13, *) {
+                if let tz = locale.timeZone {
+                    timeZone = tz
+                } else {
+                    timeZone = TimeZone(secondsFromGMT: 0)!
+                }
+            } else {
+                timeZone = TimeZone(secondsFromGMT: 0)!
+            }
+        }
+        
+        if let calID = request.calendar {
+            guard let c = self.calendars[calID] else {
+                throw Abort(.badRequest, reason: "Unknown calendar: '\(calID)'")
+            }
+            calendar = c
+            calendarID = calID
+        } else {
+            calendar = locale.calendar
+            calendarID = "\(calendar.identifier)"
+        }
+        
+        let formatter = DateFormatter()
+        formatter.calendar = calendar
+        formatter.timeZone = timeZone
+        formatter.locale = locale
+        
+        switch request.format {
+            case .date(let date):
+                formatter.dateStyle = date.dateFormatterStyle
+                formatter.timeStyle = .none
+            case .time(let time):
+                formatter.dateStyle = .none
+                formatter.timeStyle = time.dateFormatterStyle
+            case .dateAndTime(let date, let time):
+                formatter.dateStyle = date.dateFormatterStyle
+                formatter.timeStyle = time.dateFormatterStyle
+            case .template(let string):
+                formatter.dateFormat = DateFormatter.dateFormat(fromTemplate: string, options: 0, locale: locale)
+            case .raw(let string):
+                formatter.dateFormat = string
+        }
+        
+        let date: Date
+        if let timestamp = request.timestamp {
+            date = Date(timeIntervalSince1970: timestamp)
+        } else {
+            date = Date()
+        }
+        
+        let formatted = formatter.string(from: date)
+        
+        return FormatResponse(id: request.id,
+                              calendar: calendarID,
+                              timeZone: timeZone.identifier,
+                              locale: locale.identifier,
+                              timestamp: date.timeIntervalSince1970,
+                              format: formatter.dateFormat,
+                              value: formatted)
+    }
```

Comment:
```suggestion
        return FormatResponse(
            id: request.id,
            calendar: calendarID,
            timeZone: timeZone.identifier,
            locale: locale.identifier,
            timestamp: date.timeIntervalSince1970,
            format: formatter.dateFormat,
            value: formatted
        )
    }
```

minor formatting nit